docs(contributing): add a contributing guide

nicolethoen opened the pull request:
Closes https://github.com/patternfly/pf-roadmap/issues/254

patternfly-build commented:
Preview: https://pf-pr-7585.surge.sh

A11y report: https://pf-pr-7585-a11y.surge.sh

srambach commented:
Is this replacing the contribution guide and coding guidelines at https://github.com/patternfly/patternfly/tree/main/patternfly-docs/site/pages? Also, we have those files linked from the workspace, so we should pick one and point to it from there and org.

nicolethoen commented:
> Is this replacing the contribution guide and coding guidelines at https://github.com/patternfly/patternfly/tree/main/patternfly-docs/site/pages? Also, we have those files linked from the workspace, so we should pick one and point to it from there and org.

Oh I didn't see this contribution guide - in the react Repo it was at the root. I'll rewrite this and integrate it with the files we already have. sorry about that.

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -1,81 +1,46 @@
-### Common modifier class names
+**2. Find your first contribution:**
+- Browse [good first issues](https://github.com/patternfly/patternfly/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+- Look for [help wanted](https://github.com/patternfly/patternfly/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issues
```
This link says that's an invalid label - I don't think we have that one in core. @mcoker do we want to add that?

mcoker replied:
I'm not sure, we aren't actively grooming and adding tags like this to issues so I'm inclined just to remove this line. IMO "good first issue" conveys what "help wanted" would.

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -1,81 +1,46 @@
+**2. Find your first contribution:**
+- Browse [good first issues](https://github.com/patternfly/patternfly/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+- Look for [help wanted](https://github.com/patternfly/patternfly/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issues
+- Check the [project board](https://github.com/orgs/patternfly/projects) for approved work
```
Do we want to go straight to either the roadmap or the issues board? Instead of "approved work" what about "current roadmap" or "currently planned issues" or something like that?

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -1,81 +1,46 @@
-Modifier classes help us to create variations of blocks. Reuse names as much as possible to avoid confusion.
+**3. Get help:**
+- Join our [Slack community](https://patternfly.slack.com/) - `#patternfly-core` channel
+- Read our [development guidelines](https://core-staging.patternfly.org/guidelines)
```
This will need to be updated to pf-core-staging now

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -1,81 +1,46 @@
-| `pf-m-gutter`   | Adds vertical (if applicable) and horizontal gutters to the element |
+## Code of Conduct
 
+All contributors and participants agree to abide by the [PatternFly Code of Conduct](../../../CODE_OF_CONDUCT.md). Please read it before contributing.
```
This URL is broken in the preview, is the relative path correct?

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -1,81 +1,46 @@
 
+All contributors and participants agree to abide by the [PatternFly Code of Conduct](../../../CODE_OF_CONDUCT.md). Please read it before contributing.
 
 ## Pull request guidelines
```
In the last bullet of this section, can the numbered items be formatted as a list?

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -92,3 +57,72 @@ In order to streamline reviews and set expectations, the following should be exp
 
+### Submitting Pull Requests
+
+1. **Create a descriptive title** following conventional commits
```
Can we explain "conventional commits" or link to wherever we say what that is? (if we do...)

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -92,3 +57,72 @@ In order to streamline reviews and set expectations, the following should be exp
+yarn prettier
+
+# Build the project
+yarn build
```
Build isn't really necessary for testing your changes, but what if we include it (i think it's good to have the command listed) add a step and link to instructions for running the visual regression tests? (which are in a google doc at the moment). And/or - move it above `yarn a11y` because I think it's required for that step?

srambach commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -92,3 +57,72 @@ In order to streamline reviews and set expectations, the following should be exp
+
+- **Security review:** Pay special attention to security implications of AI-generated code, especially when dealing with user input, authentication, or data handling.
+- **Testing:** Ensure that AI-generated code is thoroughly tested and meets all existing project standards for testing coverage.
+- **Code style:** Verify that AI-generated code follows PatternFly's coding standards and style guidelines as outlined in this document.
```
Link to guidelines doc?

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -1,29 +1,41 @@
+
 Please enforce these guidelines at all times. Small or large, call out what's incorrect.
 
 Every line of code should appear to be written by a single person, no matter the number of contributors.
```
This always amused me - does it really make sense?

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -1,29 +1,41 @@
 Every line of code should appear to be written by a single person, no matter the number of contributors.
 
-This set of rules generate some constraints and conventions. If you run into instances where a convention isn’t obvious or a solution could be handled in a few different ways, contact the PatternFly community, have a conversation about how to handle it and update this guidelines when needed.
+This set of rules generate some constraints and conventions. If you run into instances where a convention isn't obvious or a solution could be handled in a few different ways, contact the PatternFly community, have a conversation about how to handle it and update this guidelines when needed.
```
This paragraph seems awkward and the grammar isn't very good - maybe @edonehoo would have a suggestion?

mcoker commented on patternfly-docs/site/pages/contribution.md:
```diff
@@ -92,3 +57,72 @@ In order to streamline reviews and set expectations, the following should be exp
+yarn lint
+
+# Run Prettier
+yarn prettier
```
We don't use prettier. We have a [yarn format](https://github.com/patternfly/patternfly/blob/407bf6224f1b831090a80065302c36d99d77241c/package.json#L23) script that invokes prettier, but AFAIK no one uses it.

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +46,111 @@ Components are modular and independent structures concerned with how a thing loo
+As a general rule, create an extension to an element with BEM modifiers if it's a change of color or scale. If the change generates a new entity, then create a new component.
 
-Repetition is better than the wrong abstraction.
+**Repetition is better than the wrong abstraction.**
```
I don't think this needs to be bold.

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +46,111 @@ Components are modular and independent structures concerned with how a thing loo
+- **Utilities**: `.pf-v6-u-utility-name--modifier`
+- **Prefix all PatternFly classes** with `.pf-v6-`
+
+### Variables System
```
```suggestion
### Variable System
```

mcoker commented:
For this block - https://github.com/patternfly/patternfly/blob/d4085afe6657116d8122ff37be0b0b176681a299/README.md?plain=1#L12-L17

I wonder if it's worth calling out other files and adding info to the existing bullet points. WDYT? Also all of these styles are available via `.css` and `.scss`, so not sure if we need separate bullet points for both, or if we can just mention something like "If you want the sass version of any of these stylesheets, just specify `.scss` as the suffix instead of `.css`"

* `@patternfly/patternfly/patternfly-base.css` - base styles. Global styles, reset/normalize styles, global variables. These styles are required for patternfly styles to work properly.
* `@patternfly/patternfly/patternfly.css` - this is `patternfly-base.css`, as well as all component and layout styles. Does not include utility styles.
* `@patternfly/patternfly/patternfly-addons.css` - all utility class styles. These do not come with `patternfly.css`, and must be imported manually.
* `@patternfly/patternfly/utilities/[UtilityName]/[utility-name].css` - individual utility class styles if you prefer only use one or two, and don't want to import all of them.
* `@patternfly/patternfly/patternfly-charts.css` - required if using dark theme with [PatternFly charts](https://www.patternfly.org/charts/about-charts).

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -104,20 +159,20 @@ The second layer is scoped to themeable component custom properties. This setup
   - `--pseudo-element` is one of either `before` or `after`.
   - `--child`, `--tag`, or `--c-component` refers to a child element. It could be a tag or component name, like `--c-menu` or `--svg`, or it could use `--child` to refer to any child element. If any modifiers, states, breakpoints, or pseudo-elements are on the child, include those after this portion of the name.
-- The value of a component variable is **always** defined by a global variable.
+- The value of a component variable is **always** defined by a CSS token.
```
This is not strictly true; there are instances where a component variable is assigned something like `transparent`, a grid area name, or other explicit value. Also, "CSS token" isn't the terminology I'd use.
```suggestion
- The value of a component variable should be a design token.
```
@mcoker what do you think?

mcoker replied:
Yeah I'd probably just remove this bullet point, too. I think the point it's trying to make is to not make up/hard-code colors , font-sizes, border-widths... all of the stuff we have tokens for. But you're right - we use lots of values that are not tokens.

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -104,20 +159,20 @@ The second layer is scoped to themeable component custom properties. This setup
---pf-v6-c-alert--Padding: var(--pf-global--spacer--xl);
---pf-v6-c-alert--hover--BackgroundColor: var(--pf-global--BackgroundColor--200);
---pf-v6-c-alert__title--FontSize: var(--pf-global--FontSize--2xl);
+// Component scoped variables are always defined by CSS tokens
```
```suggestion
// Component scoped variables are defined by design token variables
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +190,120 @@ For example:
+
+If a situation arises where a value needs entering into the style sheets that isn't already defined in the CSS tokens this should serve as a red flag to you.
+
+In the case where a 'magic' value needs entering, ensure a comment is added on the line above to explain its relevance.
```
```suggestion
In the case where a 'magic' value needs to be used, ensure a comment is added to explain its relevance.
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +46,111 @@ Components are modular and independent structures concerned with how a thing loo
+  - and a `state` is something like `hover`, or `clicked`.
+- They are concepts, never tied to an element or component. This is incorrect: `--pf-t--global--h1--font--size`, this is correct: `--pf-t--global--font--size--800`.
+
+For example a CSS token setup would look like:
```
for all the instances of "CSS token" I'd call them "design tokens" or "design token CSS variables"

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +190,120 @@ For example:
+  // mobile styles
+
+  // Styles for small view ports and up
+  @media (min-width: var(--pf-t--global--breakpoint--sm)) {
```
I think this needs to be a SCSS variable instead of a CSS variable.

mcoker replied:
Yeah @nicolethoen for context, CSS vars don't work in `@media` queries currently. On the dev side (for this doc), we'll use the SCSS vars below. If you're a product, you'd use:
* SCSS vars if you have access to them // preferred, will always match what PF uses for a breakpoint
* react-token values for the breakpoints // preferred, same as above
* literal size values (eg, `36rem` or `576px`)

https://github.com/patternfly/patternfly/blob/f81a8c499fe1bbbe568470de45ae1d555454e4c9/src/patternfly/sass-utilities/scss-variables.scss#L29-L35

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +190,120 @@ For example:
+
+There is currently a bug in cssnano ([issue #64 on postcss-calc](https://github.com/postcss/postcss-calc/issues/64)) that causes nested `calc()` CSS functions to be removed. So a function like `calc(5 * calc(3 - 1))` becomes `calc(5 * 3 - 1)`. It's worth noting that this problem only impacts our distribution package. Nested `calc()` functions work fine in the development environment.
+
+PatternFly developers should avoid nested `calc()` CSS functions until this bug is resolved and the package is updated in the [patternfly repository](https://github.com/patternfly/patternfly). If you're interested in following this issue, you can do so in [issue #1295 on patternfly](https://github.com/patternfly/patternfly/issues/1295).
```
I _think this_ issue has been resolved so we can remove this.

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -238,125 +365,94 @@ Limit the use of shorthand declarations to instances where you must explicitly s
 .element {
-  margin-bottom: 10px;
-  background-color: #f00;
+  margin-bottom: var(--pf-t--global--spacer--sm);
```
Let's update this to use logical directions
```suggestion
  margin-block-end: var(--pf-t--global--spacer--sm);
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
````diff
@@ -135,34 +190,120 @@ For example:
+}
+```
+
+3. **States, pseudo-classes and pseudo-elements**
````
```suggestion
3. **States, pseudo-classes, and pseudo-elements**
```
Oxford comma is a hill I'll die on. :-)

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +190,120 @@ For example:
+  &:hover {
+    background: var(--pf-v6-c-button--hover--Background);
+  }
+}
```
Should we add a pseudoelement then, since the heading mentions it but it's not shown?

nicolethoen replied:
Could you provide an example of that? I wouldn't be sure what to add.

srambach replied:
Sure, something like
`// --block--pseudoelement--PropertyCamelCase
.pf-v6-c-alert {
  &::after {
    border-color: var(--pf-v6-c-alert--after--BackgroundColor);
  }
}`

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +190,120 @@ For example:
+
+#### @mixin and @extend
+
+Since our components are isolated and modular try to avoid `@mixin` and `@extend` because they generate a dependency.
```
We have a number of mixins, so I'm not sure if we want to include this?

srambach commented on patternfly-docs/site/pages/contribution.md:
````diff
@@ -92,3 +57,72 @@ In order to streamline reviews and set expectations, the following should be exp
+    // Optimized CSS properties based on AI suggestions
+    --pf-v6-c-component-name--FontSize: var(--pf-v6-global--FontSize--md);
+  }
+  ```
````
suggested edit:
### Mark code with substantial AI-generated portions

Nontrivial and substantial AI-generated or AI-assisted content should be marked.

- **In a commit message, or in a pull request/merge request description field,** identify the code assistant that you used, perhaps elaborating on how it was used. You may wish to use a trailer like "Assisted-by:" or "Generated-by:". For example:

  ```
  Assisted-by: GitHub Copilot
  Generated-by: ChatGPT for initial component structure
  ```

- In the case of entire generated files, **in a source file comment,** indicate the use of the code assistant. For example:

  ```scss
  // Generated by GitHub Copilot
  // AI-assisted implementation with human review and modifications
  .pf-v6-c-component-name {
    // Optimized CSS properties based on AI suggestions
    --pf-v6-c-component-name--FontSize: var(--pf-v6-global--FontSize--md);
  }
  ```


@rebeccaalpert wdyt?

srambach reviewed (commented):
Apologies for the big number of individual comments! I realize a few of these are out of scope of just reorganizing the contribution and guidelines. It's been a long while since we revisited these, so it's probably time for a good review.

nicolethoen committed:
docs(contributing): update contribution guides

nicolethoen committed:
docs(contributing): link to PatternFly organizations AI guidance

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -1,29 +1,39 @@
 
 Layouts are the containers that allow for organizing and grouping its immediate children on the page.
 
 - A layout never imposes padding or element styles on its children.
```
This is actually false - we do impose spacing on flex children, for example.
```suggestion
- A layout never imposes element styles on its children.
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
 The parent container of a component is prefixed with `-c` (after the PatternFly prefix `pf-` and version `v6-`), for example: `.pf-v6-c-alert` or `.pf-v6-c-button`.
 
-### When to create a new component
+#### When to Create a New Component
```
Don't we use sentence case everywhere?
```suggestion
#### When to create a new component
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
 Repetition is better than the wrong abstraction.
 
-## Additional features
+#### Additional Features
```
```suggestion
#### Additional features
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
+3. **Support keyboard navigation** where applicable
+4. **Work across supported browsers** (latest 2 versions of Chrome, Firefox, Safari, Edge)
+5. **Be responsive** and work on mobile devices
+6. **Support theming** through CSS custom properties
```
```suggestion
6. **Support theming** through design tokens
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
+  - a `property` is something like `size` or `color`.
+  - a `modifier` is something like `100`, `200`, or `sm`.
+  - and a `state` is something like `hover`, or `clicked`.
+- They are concepts, never tied to an element or component. This is incorrect: `--pf-t--global--h1--font--size`, this is correct: `--pf-t--global--font--size--800`.
```
```suggestion
- They are concepts, never tied to an element or component. 
- Incorrect: `--pf-t--global--h1--font--size`
- Correct: `--pf-t--global--font--size--800`
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
````diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
+--pf-t--color--blue--10: #e0f0ff;
+--pf-t--color--blue--30: #92c5f9;
+--pf-t--color--blue--50: #0066cc;
 ```
````
```suggestion
Furthermore, design tokens have been grouped into palette, base, and semantic tokens. 
- Palette tokens are fundamental values like palette colors.
- Base tokens create a family of acceptable values for a concept like danger status.
- Semantic tokens assign meaningful relationships to tokens, like borders for things with danger status.

For example design tokens for setting up danger status styling would look like:

```scss
// Palette tokens
  --pf-t--color--red-orange--50: #f0561d;
  --pf-t--color--red-orange--60: #b1380b;
  --pf-t--color--red-orange--70: #731f00;

// Base tokens
  --pf-t--global--color--status--danger--100: var(--pf-t--color--red-orange--60);
  --pf-t--global--color--status--danger--200: var(--pf-t--color--red-orange--70);
  --pf-t--global--color--status--danger--300: var(--pf-t--color--red-orange--80);

// Semantic tokens 
  --pf-t--global--border--color--status--warning--clicked: var(--pf-t--global--color--status--warning--300);
  --pf-t--global--border--color--status--warning--default: var(--pf-t--global--color--status--warning--200);
  --pf-t--global--border--color--status--warning--hover: var(--pf-t--global--color--status--warning--300);

```
```

@mcoker @lucia does this make sense?

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -34,68 +44,111 @@ Components are modular and independent structures concerned with how a thing loo
-### Component variables
+#### Component variables
 
 The second layer is scoped to themeable component custom properties. This setup allows for consistency across components, generates a common language between designers and developers, and gives granular control to authors. The rules are as follows:
```
```suggestion
The second layer is scoped to customizable component custom properties. This setup allows for consistency across components, generates a common language between designers and developers, and gives granular control to authors. The rules are as follows:
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -104,20 +157,19 @@ The second layer is scoped to themeable component custom properties. This setup
+  - `--breakpoint` is a media query breakpoint such as `sm` for `--pf-t--global--breakpoint--sm`.
   - `--pseudo-element` is one of either `before` or `after`.
   - `--child`, `--tag`, or `--c-component` refers to a child element. It could be a tag or component name, like `--c-menu` or `--svg`, or it could use `--child` to refer to any child element. If any modifiers, states, breakpoints, or pseudo-elements are on the child, include those after this portion of the name.
-- The value of a component variable is **always** defined by a global variable.
```
I'd leave this and say "The value of a component variable should refer to a **semantic design token**."

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -104,20 +157,19 @@ The second layer is scoped to themeable component custom properties. This setup
+// Component scoped variables are defined by design token variables
+--pf-v6-c-alert--Padding: var(--pf-t--global--spacer--400);
+--pf-v6-c-alert--hover--BackgroundColor: var(--pf-t--global--background--color--200);
+--pf-v6-c-alert__title--FontSize: var(--pf-t--global--font--size--700);
```
```suggestion
// Component scoped variables are defined by semantic design token variables
--pf-v6-c-alert--FontSize: var(--pf-t--global--font--size--body--default);
--pf-v6-c-alert__title--Color: var(--pf-t--global--text--color--regular);
--pf-v6-c-alert--m-danger--BorderColor: var(--pf-t--global--border--color--status--danger--default);
```

srambach commented on patternfly-docs/site/pages/guidelines.md:
```diff
@@ -135,34 +187,114 @@ For example:
-If a situation arises where a value needs entering into the style sheets that isn't already defined in the global variables this should serve as a red flag to you.
+#### Nested calc() functions
+
+There is currently a bug in cssnano ([issue #64 on postcss-calc](https://github.com/postcss/postcss-calc/issues/64)) that causes nested `calc()` CSS functions to be removed. So a function like `calc(5 * calc(3 - 1))` becomes `calc(5 * 3 - 1)`. It's worth noting that this problem only impacts our distribution package. Nested `calc()` functions work fine in the development environment.
```
This issues has been resolved.

srambach commented on README.md:
```diff
@@ -68,11 +73,11 @@ When making visual changes to a full page example, new example preview screensho
+[For more information on using CSS, Sass, and handlebars, refer to our guidelines.](https://pf-core-staging.patternfly.org/guidelines)
 
 ### Custom icon guidelines
 [For more information on custom icons, refer to our guidelines.](https://pf-core-staging.patternfly.org/adding-custom-icons)
```
```suggestion
For more information on custom icons, refer to our [custom icon guidelines](https://pf-core-staging.patternfly.org/adding-custom-icons).
```

srambach commented on README.md:
```diff
@@ -68,11 +73,11 @@ When making visual changes to a full page example, new example preview screensho
-### CSS/Sass guidelines
-[For more information on using CSS and Sass, refer to our guidelines.](https://pf-core-staging.patternfly.org/guidelines)
+### Development guidelines
+[For more information on using CSS, Sass, and handlebars, refer to our guidelines.](https://pf-core-staging.patternfly.org/guidelines)
```
```suggestion
For more information on using CSS, Sass, and handlebars, refer to our [development guidelines](https://pf-core-staging.patternfly.org/guidelines).
```

srambach commented on README.md:
```diff
@@ -68,11 +73,11 @@ When making visual changes to a full page example, new example preview screensho
-### Handlebars guidelines 
-[For information on how to contribute, refer to our guidelines.](https://pf-core-staging.patternfly.org/contribution)
+### Contribution guide 
+[For information on how to contribute, refer to our guide.](https://pf-core-staging.patternfly.org/contribution)
```
```suggestion
For information on how to contribute, refer to our [contribution guide](https://pf-core-staging.patternfly.org/contribution).
```

srambach reviewed (commented):
Made another pass through. Honestly, the guidelines should probably be broken up or at least have jumplinks to the sections. 😵‍💫  But for now I made some corrections to token usage and a few other nits.

nicolethoen committed:
fix(guidelines): address PR feedback

nicolethoen commented:
I addressed your comments! Thank you so much!

> Honestly, the guidelines should probably be broken up or at least have jumplinks to the sections. 😵‍💫 But for now I made some corrections to token usage and a few other nits.

I think you all can absolutely do an even more substantial rewrite or reorganization in the future! They've been so out of date for so long 😆

nicolethoen committed:
fix(custom-icons): fix broken link

nicolethoen committed:
fix(guidelines): add pseudo element example

mcoker reviewed (approved):
Just some broken links it looks like. These pages need a good overhaul, I'll spare any comments about that 😅

* slack link should be https://patternfly.slack.com/ https://github.com/patternfly/patternfly/blob/d102c68cc02aef2c8bf4919adb92756d747b7bf7/patternfly-docs/site/pages/contribution.md?plain=1#L46
* a11y link doesn't work. Should probably go to https://www.patternfly.org/accessibility/about-accessibility? https://github.com/patternfly/patternfly/blob/d102c68cc02aef2c8bf4919adb92756d747b7bf7/patternfly-docs/site/pages/guidelines.md?plain=1#L43
* link doesn't work, should be https://cssguidelin.es/#80-characters-wide https://github.com/patternfly/patternfly/blob/d102c68cc02aef2c8bf4919adb92756d747b7bf7/patternfly-docs/site/pages/guidelines.md?plain=1#L317
* TOC link doesn't work, should be https://cssguidelin.es/#table-of-contents https://github.com/patternfly/patternfly/blob/d102c68cc02aef2c8bf4919adb92756d747b7bf7/patternfly-docs/site/pages/guidelines.md?plain=1#L311

nicolethoen committed:
fix(links): update broken links

patternfly-build commented:
:tada: This PR is included in version 6.3.0-prerelease.54 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v6.3.0-prerelease.54)
- [npm package (@prerelease dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/6.3.0-prerelease.54)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
